packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -396,6 +517,25 @@ abstract class ImageStreamCompleter with Diagnosticable {
     }
   }
 
+  /// Adds a listener callback that is called only called if there are listeners
```

Comment:
I am not sure I understand what this paragraph is saying. When should I add a regular listener? When a passive one?

Reply:
The image cache wants to listen without actually requesting more frames. Maybe there's a better name for this.